### PR TITLE
[WIP] Add Dockerfile for `modelmesh-minio-examples` image

### LINKED ISSUE #297: Add Dockerfile for `modelmesh-minio-examples` image
Add the `Dockerfile` for the [`modelmesh-minio-examples` image](https://hub.docker.com/layers/kserve/modelmesh-minio-examples/v0.9.0/image/sha256-4c5185e400d83ff22c3c3ac165ab43cb0c5283843782cdfb73a89c9dac46ddd2?context=explore) to the `modelmesh-serving` repo.

```Dockerfile
From quay.io/cloudservices/minio:RELEASE.2021-06-17T00-10-46Z.hotfix.35a0912ff
EXPOSE 9000
RUN mkdir -p /data1/modelmesh-example-models
COPY sklearn /data1/modelmesh-example-models/sklearn/
COPY lightgbm /data1/modelmesh-example-models/lightgbm/
COPY onnx /data1/modelmesh-example-models/onnx/
COPY pytorch /data1/modelmesh-example-models/pytorch/
COPY xgboost /data1/modelmesh-example-models/xgboost/
COPY tensorflow /data1/modelmesh-example-models/tensorflow/
COPY simple_string /data1/modelmesh-example-models/tensorflow/simple_string/
COPY keras /data1/modelmesh-example-models/keras/
# for ftv image only
COPY fvt /data1/modelmesh-example-models/fvt/
COPY <your files> 
```

And the instructions to build the image, i.e.:

```Bash
docker build -t <image registry>/<org>/<repo name>:<tag>
docker push <image registry>/<org>/<repo name>:<tag>
```

Once we have the Dockerfile we should be able to add a Gtihub action to push it whenever it gets updated

Some model files are large and are managed the IBM COS. Maybe we can pull them into the Docker build from their respective sources … i.e. have a build script that copies the large files to a local temp dir before the docker build command starts and then deletes the large temp files once the docker build finished.

The Dockerfile itself could go under a new directory or something called `examples` (or better)

As a workaround for this current `v0.10.0` release I retagged the existing `latest` image on DockerHub directly.
```Bash
docker manifest create kserve/modelmesh-minio-examples:v0.10.0 kserve/modelmesh-minio-examples:latest
docker manifest push kserve/modelmesh-minio-examples:v0.10.0
```

/assign @tedhtchang

---

**UPDATE 2/1/2023:** We now have the [kserve/modelmesh-minio-examples](https://github.com/kserve/modelmesh-minio-examples) repository dedicated to maintain and build the ModelMesh MinIO Examples Docker image(s):

- https://github.com/kserve/modelmesh-minio-examples/blob/main/Dockerfile

### PR DESCRIPTION
**Pull request by tedhtchang**
Closes #297 #302
Signed-off-by: ted chang <htchang@us.ibm.com>

#### Motivation

Provide instructions to build a MinIO model examples image

#### Modifications
N/A
#### Result
N/A

### CODE REVIEW

**Comment by ckadner** [comment]
Hi @tedhtchang -- thank you for your PR. Do you know where all those large model files originate from? Since they are so large, I think it would be great if we don't have to keep a copy of those files in the repo, instead, maybe they could be copied/downloaded from their original source at the time we build the image? Or whenever we need them for testing. And delete them afterwards.

**Comment by rafvasq** [comment]
Thanks @tedhtchang this is really helpful. 

Probably a question for @njhill and @ckadner, I'm wondering whether `fvt` would be the correct directory to host both Dockerfiles. Maybe it'd be a new `quickstart` directory for the examples-only Dockerfile while the FVT-related image stays here?

**Comment by njhill** [comment]
Yes thanks @tedhtchang. I think we should add just the Dockerfile(s) for now while we figure out where to host the larger model files, maybe we could do that in a separate PR so that we keep this branch with all the files until we figure that out.

I agree with @rafvasq that the existing `fvt` dir probably isn't the best place for this to live. Probably it makes sense to have a new top-level dir called something like `mino_build`, since I can't really see anywhere else that it would fit very well.

**Comment by ckadner** [comment]
Hi Ted,

I moved the files to into a new `minio-examples` folder at the project root level, for now.

Although I don’t think we should add this to the `modelmesh-serving` repo — a 600 MB addition is just too large. 

I think we create a new repo soon `modelmesh-minio-examples` with it's own CI/CD and figure out a way to offload the model files either to COS buckets with public read-access (or even credentialed access for MM maintainers only) or we use another mechanism like [Git LFS](https://docs.github.com/en/billing/managing-billing-for-git-large-file-storage/about-billing-for-git-large-file-storage#about-billing-for-git-large-file-storage) for **L**arge **F**ile **S**torage as @njhill has suggested

File: minio_examples/Dockerfile [hunk 1]
```diff
@@ -17,18 +17,26 @@ FROM quay.io/cloudservices/minio:RELEASE.2021-06-17T00-10-46Z.hotfix.35a0912ff a
...
+COPY lightgbm /data1/modelmesh-example-models/lightgbm/
+COPY onnx /data1/modelmesh-example-models/onnx/
+COPY pytorch /data1/modelmesh-example-models/pytorch/
+COPY xgboost /data1/modelmesh-example-models/xgboost/
+COPY tensorflow /data1/modelmesh-example-models/tensorflow/
+COPY keras /data1/modelmesh-example-models/keras/
+
+RUN useradd -u 1000 -g 0 modelmesh
+
+RUN chown -R 1000:0 /data1
```
Comment (ckadner) [inline minio_examples/Dockerfile:24]: as Nick pointed out, doing a `chown` after the `COPY` layers causes the final Docker image to be about 100 MB larger. Reverting to `COPY --chown 1000:0 ...` in next commit

Comment (ckadner) [reply minio_examples/Dockerfile:24]: https://hub.docker.com/r/kserve/modelmesh-minio-examples/tags
https://hub.docker.com/r/kserve/modelmesh-minio-dev-examples/tags

**Comment by ckadner** [comment]
@rafvasq -- I pushed a new `modelmesh-minio-dev-examples:latest` image with the FVT models you added

https://hub.docker.com/r/kserve/modelmesh-minio-dev-examples/tags

**Comment by ckadner** [comment]
I created a new repository https://github.com/kserve/modelmesh-minio-examples/ from the commits in this PR with a few modifications to 
- remove the biggest unused models:
  - fvt/onnx/onnx-mobilenet         (14 MB)
  - fvt/onnx/onnx-pytorch-converted (4.5 MB)
  - fvt/onnx/onnx-test-model        (38 MB)
  - fvt/tensorflow/tfmnist              (78 MB)
  - fvt/tensorflow/tfmnistnew       (39 MB) 
- removing any duplicate models in favor of doing [`COPY`s in the Dockerfile](https://github.com/kserve/modelmesh-minio-examples/commit/c337802702b5b4190593978919bc6e56061f0873):
  - fvt/keras                           (6.5 MB)
  - fvt/pytorch/pytorch-mar-dup         (4.6 MB)
  - fvt/tensorflow/keras-mnist          (6.5 MB)
  - fvt/tensorflow/keras-mnistnew       (6.5 MB)
  - fvt/tensorflow/mnist-dup.savedmodel (39 MB)
  - fvt/tensorflow/mnist.savedmodel     (39 MB)
  - tensorflow/mnist.savedmodel         (39 MB)

**Comment by ckadner** [comment]
@tedhtchang -- thank you again for this PR. It was really helpful to get the new  https://github.com/kserve/modelmesh-minio-examples/ repo seeded. If you don't mind keeping you fork and branch alive for the coming weeks so we can go back to it in case some of my trimming down was over eager ;-)

**Comment by tedhtchang** [comment]
> @tedhtchang -- thank you again for this PR. It was really helpful to get the new https://github.com/kserve/modelmesh-minio-examples/ repo seeded.

Good good. Thanks.

> If you don't mind keeping your fork and branch alive for the coming weeks so we can go back to it in case some of my trimming down was over eager ;-)

I don't mind.

**Pull request closed without merging**